Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM ruby:2.5
+
+# Add yarn apt repository
+# https://classic.yarnpkg.com/en/docs/install#debian-stable
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list
+
+# Install system packages
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y \
+      build-essential \
+      imagemagick \
+      libarchive-dev \
+      libffi-dev \
+      libmagickwand-dev \
+      libpq-dev \
+      libsasl2-dev \
+      libxml2-dev \
+      libxslt1-dev \
+      locales \
+      nodejs \
+      osmosis \
+      phantomjs \
+      postgresql-client \
+      ruby-dev \
+      yarn && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Setup app location
+RUN mkdir -p /app
+WORKDIR /app
+
+# Install Ruby packages
+ADD Gemfile Gemfile.lock /app/
+RUN bundle install
+
+# Install NodeJS packages
+ADD package.json yarn.lock /app/
+RUN yarn
```

Comment:
@gravitystorm I realize you've been consistently asking for using the `bundle exec` form of commands for consistency _(which I appreciate)_, but I'll try to convince you of this exception... 

I toiled with getting this to work via `bundle exec rake db:migrate`, but I couldn't get it to run without adding a big chunk of the repository into the image during build. It ended up adding a bunch of extra files to the image *just* so that I could run `bundle exec rake db:migrate` instead `yarn` directly.  I believe the result is the same in the end so I went with the simpler option.

That said, perhaps someone else could compile a small list of files necessary to successfully run `rake` commands during the image build?

Reply:
You're already done `bundle install` which should install everything `bundle exec` might need - if you don't use `bundle exec` then you have no idea if things are using the expected modules, or even if they will work at all.